code/modules/mob/living/life.dm

```diff
@@ -33,75 +41,76 @@
+			handle_mutations_and_radiation(delta_time, times_fired)
 
 		if(stat != DEAD)
 			//Breathing, if applicable
-			handle_breathing(times_fired)
+			handle_breathing(delta_time, times_fired)
 
-		handle_diseases()// DEAD check is in the proc itself; we want it to spread even if the mob is dead, but to handle its disease-y properties only if you're not.
+		handle_diseases(delta_time, times_fired)// DEAD check is in the proc itself; we want it to spread even if the mob is dead, but to handle its disease-y properties only if you're not.
 
-		handle_wounds()
+		handle_wounds(delta_time, times_fired)
 
 		if (QDELETED(src)) // diseases can qdel the mob via transformations
 			return
 
 		if(stat != DEAD)
 			//Random events (vomiting etc)
-			handle_random_events()
+			handle_random_events(delta_time, times_fired)
 
 		//Handle temperature/pressure differences between body and environment
 		var/datum/gas_mixture/environment = loc.return_air()
 		if(environment)
-			handle_environment(environment)
+			handle_environment(environment, delta_time, times_fired)
 
-		handle_gravity()
+		handle_gravity(delta_time, times_fired)
 
 		if(stat != DEAD)
-			handle_traits() // eye, ear, brain damages
-			handle_status_effects() //all special effects, stun, knockdown, jitteryness, hallucination, sleeping, etc
+			handle_traits(delta_time, times_fired) // eye, ear, brain damages
+			handle_status_effects(delta_time, times_fired) //all special effects, stun, knockdown, jitteryness, hallucination, sleeping, etc
 
-	handle_fire()
+	handle_fire(delta_time, times_fired)
 
 	if(machine)
 		machine.check_eye(src)
 
 	if(stat != DEAD)
 		return 1
 
-/mob/living/proc/handle_breathing(times_fired)
+/mob/living/proc/handle_breathing(delta_time, times_fired)
 	return
 
-/mob/living/proc/handle_mutations_and_radiation()
+/mob/living/proc/handle_mutations_and_radiation(delta_time, times_fired)
 	radiation = 0 //so radiation don't accumulate in simple animals
 	return
 
-/mob/living/proc/handle_diseases()
+/mob/living/proc/handle_diseases(delta_time, times_fired)
 	return
 
-/mob/living/proc/handle_wounds()
+/mob/living/proc/handle_wounds(delta_time, times_fired)
 	return
 
-/mob/living/proc/handle_random_events()
+/mob/living/proc/handle_random_events(delta_time, times_fired)
 	return
 
 // Base mob environment handler for body temperature
-/mob/living/proc/handle_environment(datum/gas_mixture/environment)
+/mob/living/proc/handle_environment(datum/gas_mixture/environment, delta_time, times_fired)
 	var/loc_temp = get_temperature(environment)
+	var/temp_delta = loc_temp - bodytemperature
 
-	if(loc_temp < bodytemperature) // it is cold here
+	if(temp_delta < 0) // it is cold here
 		if(!on_fire) // do not reduce body temp when on fire
-			adjust_bodytemperature(max((loc_temp - bodytemperature) / BODYTEMP_DIVISOR, BODYTEMP_COOLING_MAX))
+			adjust_bodytemperature(max(max(temp_delta / BODYTEMP_DIVISOR, BODYTEMP_COOLING_MAX) * delta_time, temp_delta))
```

Comment:
One, because we already have the temperature delta and we might as well use it. Two, I thought I would protect against overshooting in case someone slows SSmobs way down.